### ISSUE TITLE
Role with implied hidden content has keyboard focus in accordion component

### ISSUE THREAD

**Issue by poojagunturu96** [middlebury/midd-frontend#655]
Some elements have a role that implies their internal structure should be hidden from assistive technologies. In other words, their sub-elements have a presentational role.

If a sub-element has keyboard focus, it should be moved outside of the presentation role. This is currently is the case with the course list and faq components. Below is a screenshot for reference:
![Screenshot 2024-09-24 at 6 24 02 PM](https://github.com/user-attachments/assets/2af94c56-4b31-49da-a0ea-016f4f81ddfe)

Siteimprove report for this issue: https://my2.siteimprove.com/Accessibility/7333/NextGen/Issue/1?conformance=&siteTargetIssueKinds=1,2&wcagVersion=21&pageSegments=&ruleName=sia-r90&ruleId=90&issueKind=1&exceptTags=1,2&siteTarget.wcagVersion=21&siteTarget.conformanceLevels=0,1,3,4&siteTarget.issueKinds=1,2
